File: app/src/main/java/com/permissionnanny/data/AppPermissionDB.java
package com.permissionnanny.data;

import java.util.ArrayList;

/**
 *
 */
public class AppPermissionDB {

    private static final String pipe = "\u0378\u0379";

    private final NannyDB mDB;

    public AppPermissionDB(NannyDB db) {
        mDB = db;
    }

    public void open() {
        mDB.open();
    }

    public ArrayList<AppPermission> getAllConfigs() {
        return mDB.findVals(null, AppPermission.class);
    }

    public void putConfig(AppPermission config) {
        mDB.put(key(config), config);
    }

    public void delConfig(String appPackage, String permission) {
        mDB.del(key(appPackage, permission));
    }

    public void delApp(String appPackage) {
        for (String key : mDB.findKeys(appPackage)) {
            mDB.del(key);
        }
    }

    public void delConfig(AppPermission config) {
        mDB.del(key(config));
    }

    private String key(AppPermission config) {
        return key(config.mAppPackageName, config.mPermissionName);
    }

    private String key(String appPackage, String permission) {
        return appPackage + pipe + permission;
    }
}

File: app/src/main/kotlin/com/permissionnanny/data/AppPermissionDB.kt
package com.permissionnanny.data

import java.util.*

/**

 */
open class AppPermissionDB(private val db: NannyDB) {

    fun open() {
        db.open()
    }

    val allConfigs: ArrayList<AppPermission>
        get() = db.findVals(null, AppPermission::class.java)

    fun putConfig(config: AppPermission) {
        db.put(key(config), config)
    }

    fun delConfig(appPackage: String, permission: String) {
        db.del(key(appPackage, permission))
    }

    fun delApp(appPackage: String) {
        for (key in db.findKeys(appPackage)) {
            db.del(key)
        }
    }

    fun delConfig(config: AppPermission) {
        db.del(key(config))
    }

    private fun key(config: AppPermission): String {
        return key(config.appPackageName, config.permissionName)
    }

    private fun key(appPackage: String, permission: String): String {
        return appPackage + pipe + permission
    }

    companion object {

        private val pipe = "\u0378\u0379"
    }
}
